src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -486,8 +480,9 @@ protected void setupChild(Configuration childConf)
             }
         }
         catch (final CheckstyleException ex) {
-            throw new CheckstyleException("cannot initialize module " + name
-                    + " - " + ex.getMessage(), ex);
+            throw new CheckstyleException(
+                    getLocalizedMessage("Checker.setupChildModule", name)
+                            + " - " + ex.getMessage(), ex);
```

Comment:
All should be in message, nobody limit you to have `{1}`.
All should be in parameters of message. No any concatenation of content in code.

Reply:
>  I was trying a rather different approach with one of the exceptions but there were conflicts

@romani , 
Actually, the change you suggested now is what I tried previously. I removed call to getMessage from hardcoded string and pointed it to .properties file as {1}
However, there were conflicts among two ci.
The `getLocalizedMessage` takes varargs as arguments, so we need to pass param Obj but the _ex.getMessage()_ returns a string (not a string object). So, without casting, we get a survivng error in `Checker / check (nullness-optional-interning) (pull_request)`
When we cast to type object (or even use string constructor invocation), we have violations with idea inspections saying the cast is redundant. I did not see a simpler way here to tackle both these ci other than pulling out ex.getMessage()
If there is any better way then I am all ears :)

Reply:
If static analysis tools blocks to make better code , we do suppression of them.

Reply:
I kept seeing in many other cases that the maintainers generally recommend not to extend surviving errors. So, I thought we won't be allowing it here too.

> If static analysis tools blocks to make better code , we do suppression of them.

One suppression coming right up....

Reply:
Checker is not fully enforced tool, I would rather sacrifice it, in favor all other, we will deal with it later on.

Main point that whole message must be in property file. All other are internal details that we can change at any time. Message is property file is kind of a contract to all users, it should not change without notifying them, user should be able to translate and change it if required.

Reply:
Please ignore this push. It is to obtain patch